server.js

```diff
@@ -1,9 +1,20 @@
+const taskRouter = require('./routes/tasks')
+const setMessage = require('./middleware/message')
+
+require('dotenv').config()
+const connectDB = require('./db/connect')
+const session = require('express-session')
+
 var express = require('express');
 var app = express();
 
 // set the view engine to ejs
 app.set('view engine', 'ejs');
 
+app.use(session({ secret: process.env.SESSION_SECRET, resave: false, saveUninitialized: true })); //set up the session
+app.use(express.urlencoded({extended: false})) //invokes express middleware to parse the data that is returned when the browser posts form results.
+app.use('/tasks', setMessage, taskRouter)//invokes the message middleware and the routes you created
```

Comment:
Good comments for context. I'd probably put them on their own line above the code that they reference, since those lines are each quite long.

Reply:
done